L1Trigger/L1THGCal/src/backend/HGCalShowerShape.cc

```diff
@@ -553,25 +556,26 @@ int HGCalShowerShape::bitmap(const l1t::HGCalMulticluster& c3d, int start, int e
     for (const auto& id_tc : triggerCells) {
       if (!pass(*id_tc.second, c3d))
         continue;
-      unsigned layer = triggerTools_.getTriggerGeometry()->triggerLayer(id_tc.second->detId());
-      if (layer > nlayers)
-        continue;
-      layers[layer] += id_tc.second->pt();
+      unsigned layer = triggerTools_.triggerLayer(id_tc.second->detId());
+      if (layer ==0 || layer > nlayers) continue;
+      layers[layer - 1] += id_tc.second->pt(); //shift by -1 because layer 0 doesn't exist
     }
   }
   double tot = 0;
-  for (unsigned i = 1; i < layers.size(); ++i)
+  for (unsigned i = 0; i < layers.size(); ++i)
     tot += layers[i];
-  int bitmap = 0;
-  for (int i = start; i <= end; i++) {
-    bitmap = bitmap << (layers[i] >= threshold);
+  uint32_t bitmap = 0;
+  for (int i = start-1; i <= end-1; i++) {
+    if ((end-i) >32) continue;
```

Comment:
- The boundary check could be moved outside the loop.
- A warning message could be added in case the specified bounds don't fit in the bitmap size.